controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -156,6 +158,50 @@ public String getScopeName() {
         return deleteStream();
     }
 
+    /**
+     * Create a subscribersRecord with provided subscriber for this Stream.
+     * @return future of operation.
+     */
+    @Override
+    public CompletableFuture<Void> createSubscribersRecord(String subscriber) {
+        ImmutableMap<String, SubscriberConfiguration> subscribers
+                = new ImmutableMap.Builder<String, SubscriberConfiguration>()
+                .put(subscriber, SubscriberConfiguration.EMPTY).build();
+        return createSubscribersDataIfAbsent(new StreamSubscribersRecord(subscribers));
+    }
+
+    public CompletableFuture<Void> updateSubscribers(final String subscriber,
```

Comment:
please add javadoc.. also mention that it can throw write conflict exceptions.

Reply:
ok. sure